packages/quantic/force-app/main/default/lwc/quanticResultAction/__tests__/quanticResultAction.test.js

```diff
@@ -0,0 +1,339 @@
+
+      expect(tooltip).not.toBeNull();
+      expect(tooltip.textContent).toBe(exampleLabelWhenHover);
+    });
+
+    it('should display the correct icon', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const iconButton = element.shadowRoot.querySelector(
+        'lightning-button-icon'
+      );
+
+      expect(iconButton).not.toBeNull();
+      expect(iconButton.iconName).toBe(exampleIconName);
+    });
+
+    it('should dispatch a click event when the lightning button is clicked', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      setupClickEventDispatchTest(element);
+
+      clickButton(element, 'lightning-button-icon');
+      expect(functionsMocks.listener).toHaveBeenCalledTimes(1);
+    });
+
+    it('should display the button properely when its in the first position', async () => {
+      order = 'first';
+      setupRegisterEventDispatchTest();
+      const element = createTestComponent();
+      await flushPromises();
+
+      const iconButton = element.shadowRoot.querySelector(
+        'lightning-button-icon'
+      );
+
+      expect(functionsMocks.listener).toHaveBeenCalledTimes(1);
+      expect(iconButton.className).toBe(
+        'result-action_button result-action_first'
+      );
+    });
+
+    it('should display the button properely when its in a middle position', async () => {
+      order = 'middle';
+      setupRegisterEventDispatchTest();
+      const element = createTestComponent();
+      await flushPromises();
+
+      const iconButton = element.shadowRoot.querySelector(
+        'lightning-button-icon'
+      );
+
+      expect(functionsMocks.listener).toHaveBeenCalledTimes(1);
+      expect(iconButton.className).toBe(
+        'result-action_button result-action_middle'
+      );
+    });
+
+    it('should display the button properely when its in the last position', async () => {
+      order = 'last';
+      setupRegisterEventDispatchTest();
+      const element = createTestComponent();
+      await flushPromises();
+
+      const iconButton = element.shadowRoot.querySelector(
+        'lightning-button-icon'
+      );
+
+      expect(functionsMocks.listener).toHaveBeenCalledTimes(1);
+      expect(iconButton.className).toBe(
+        'result-action_button result-action_last'
+      );
+    });
+
+    it('should display the button properely when its the only result action displayed', async () => {
+      order = null;
+      setupRegisterEventDispatchTest();
+      const element = createTestComponent();
+      await flushPromises();
+
+      const iconButton = element.shadowRoot.querySelector(
+        'lightning-button-icon'
+      );
+
+      expect(functionsMocks.listener).toHaveBeenCalledTimes(1);
+      expect(iconButton.className).toBe('result-action_button');
+    });
+  });
+
+  describe('when the toggle mode is enabled', () => {
```

Comment:
Same as before, I would merge some of the tests, probably all tests where the `selected` is false.